c4builder/src/3 Component Diagrams/2 Identity service/identity-service.md

```diff
@@ -1 +1,7 @@
-Placeholder
\ No newline at end of file
+**Level 3: Component diagram for *Identity* service**
+
+*Identity* service acts as a guard for entire system, managing user accounts infomation, providing authentication and allowing (or restricting) authenticated users access to parts of the system.
+
+*CQRS* (Command Query Responsibility Segregation) is used for reduced coupling between facade (controllers) and internal business logic services. 
+
+According to *CQS* (Command-query separation) principles, in the system *Queries* retrieves data from database and processes it before returning it to user (without any side effects on system state) and *Commands* performs actions that changes state of the system without returning any data.
```

Comment:
As I was writing my part of documentation I was wondering about if we can take this part, of what patterns we use, to higher level of C4 diagram or to any other document?

Reply:
This description is component (service) scoped. CQS is not a system-wide pattern, it's on per project basis. Component diagram should fully and extensively describe architecture used for designing a service. `Identity service` design shouldn't have any knowledge about service scoped architecture/flow of other services. If other services uses CQS, they should describe it as well because the only information outside of Component's scope that it has access to, should be any contracts (e.g. produced/consumed messages) or system-wide patterns/design. Moreover, there could be changes to CQS specifics (i.e. in some implementations, Commands do not return **anything** and in others, they could return some identification of created resource). Nevertheless, that's only my opinion, so feel free to object.